File: solutions/ocaml/leap/1/leap.ml
let divisible_by n year =
  year mod n = 0

let leap_year year =
  let is_divisible_by_4 = divisible_by 4 year in
  let is_century_year    = divisible_by 100 year in
  let is_divisible_by_400 = divisible_by 400 year in

  (is_divisible_by_4 && not is_century_year)
    || is_divisible_by_400

File: solutions/python/leap/1/leap.py
DIVISIBILITY_BY_4 = 4
DIVISIBILITY_BY_100 = 100
DIVISIBILITY_BY_400 = 400

def leap_year(year):
    return (
        year % DIVISIBILITY_BY_4 == 0
        and (
            year % DIVISIBILITY_BY_100 != 0
            or year % DIVISIBILITY_BY_400 == 0
        )
    )